[ONL-7891] Fix SB controls

mcibique opened the pull request:
Fixed usage of types in TS stories:
- a story written for SB 6 (`Template.bind()` style) should follow typing documentation: https://storybook.js.org/docs/6.5/vue/writing-stories/introduction
- a story written for SB 7 (StoryObj) should follow new typing documentation: https://storybook.js.org/docs/vue/writing-stories/introduction/

Not using proper types as documented disables the controls in the SB and also force us to ts-ignore `story.args` and `story.parameters` -> we want to have these typed.

All new components should use the new Component Story Format 3 format (for SB 7). Do not copy code from other components when creating new components.

And all components should be always controllable via controls, so it can be adjusted by anyone while browsing https://chameleon-ebury.vercel.app/. I have reverted some hardcoded values that used to be managed via controls.

CHANGES:
- Updated SB, Vite, Eslint for Vue and Jest. Enabled Node 18 and 20 instead of 16 which is out of LTS
- Fixed types for ec-table and ec-tooltip directive
- Fixed types in all `story.ts` files.
- Fixed destructuring of args in stories as thy are no longer an object of refs but an reactive object.
- Removed hacks introduced during SB7 migrations that were fixed by the SB team.
- Removed some hardcoded values from the stories.
- Minor tweaks in the stories: wrong indentation, enums assigned directly into the controls, fixed formatting in SFCs using new eslint rules for vue.

wolandec commented on .eslintrc.js:
do we need eburyonline commit to adjust that repo codebase accordingly these changes?

mcibique replied:
Yes, I'll do a few adjustments in EBO too, it's not many of them, I already checked that.

wolandec commented:
We need to check screenshots, if they changed after these changes.

mcibique commented on .storybook/preview.js:
```diff
@@ -4,10 +4,10 @@ import { INITIAL_VIEWPORTS } from '@storybook/addon-viewport';
-import greenTheme from '../src/styles/themes/green.css';
-import hotpinkTheme from '../src/styles/themes/hotpink.css';
-import redTheme from '../src/styles/themes/red.css';
+import bwTheme from '../src/styles/themes/b-w.css?inline';
```
I updated vite and it's now recommended: https://vitejs.dev/blog/announcing-vite4.html#importing-css-as-a-string. It complains not having inline there when starting SB.

derrick-g commented on src/components/ec-btn/ec-btn.story.ts:
On testing ec-btn when first loading the 'docs' section, the controls for icon, is-outline, is-reversed and below are not working as expected. Only when switching to the basic section, and then back do they become controllable.

derrick-g commented on src/components/ec-checkbox/ec-checkbox.story.js:
The ec-checkbox seems to have the same issue as ec-btn

derrick-g commented on src/components/ec-dropdown-search/ec-dropdown-search.story.js:
Ec-dropdown-search is presenting this error: 
`Error: falseReason is not defined
  at generateComponentSource (/assets/config-9899c9c7.js:2:1068))
  at /assets/config-9899c9c7.js:2:687
  at Array.map (<anonymous>))
  at generateComponentSource (/assets/config-9899c9c7.js:2:680))
  at /assets/config-9899c9c7.js:2:687
  at Array.map (<anonymous>))
  at generateComponentSource (/assets/config-9899c9c7.js:2:680))
  at /assets/config-9899c9c7.js:2:687
  at Array.map (<anonymous>))`

mcibique replied:
Fixed in https://github.com/Ebury/chameleon/pull/1716/commits/56162f7ddea534d8c9bdd7fcf490400b34ee3d9d. It's a bizarre bug in SB's sample code generator. It failed when trying to parse `{{ item.disabledReason }}` with error `Reason is not defined`? The item also has a property `disabled` so the generator thinks this is access to the `disabled` prop and the Reason is another variable ... WTH? I wrapped it with quotes and it works now.

derrick-g commented on src/components/ec-dropdown/ec-dropdown.story.js:
The controls don't seem available here?

derrick-g commented on src/components/ec-radio-btn/ec-radio-btn.story.ts:
None of the controls are working on this one.

derrick-g commented on src/components/ec-radio-btn-group/ec-radio-btn-group.story.ts:
The group controls are also presenting issues

derrick-g replied:
The radio-btn group is still having the same issue unfortunately

mcibique replied:
I missed the story when fixing toRefs -> watchEffect. Should be okay now.

mcibique commented:
> We need to check screenshots, if they changed after these changes.

Done, I have updated screenshots, added latest node/chrome/firefox and fixed some stories too.
